### ISSUE TITLE
Method calls (`:`) are not handled correctly if the method definition does not use `:`

### ISSUE THREAD

**Issue by raiguard** [CppCXY/emmylua-analyzer-rust#302]
![Image](https://github.com/user-attachments/assets/b2e89520-4a52-469e-92b6-f657605c996b)

Expected: `data:extend` should not throw any warnings.

Changing the definition to `data:extend(_otherdata)` fixes it, but it ought to function correctly in both cases.

The actual code can be found [here](https://github.com/wube/factorio-data/blob/master/core/lualib/dataloader.lua).

<details><summary>Code</summary>

```lua
--- @class data
data = {}
data.raw = {}
data.is_demo = false

--- @param _self data
function data.extend(_self, _otherdata)
  -- Impl
end

data:extend({
  {
    type = "item",
    name = "my-item",
  },
})


data.extend(data, {
  {
    type = "item",
    name = "my-item"
  }
})
```
</details>
